### PR TITLE
Updates the create remote cluster endpoint to require a default team id

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
Support for RemoteClusters without a default team id is in place for old servers that created those connections before v10.1. This change forbids the creation of new RemoteClusters without providing this field, and will be removed when manual invites are implemented.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-61031

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by mgdelacroix** [comment]
This PR will break the current remote cluster creation form, so I've marked it as `do not merge` until we have the matching PR merged, but it is ready for review /cc @calebroseland

**Review by lindy65 — APPROVED** [review]
Thanks @mgdelacroix - I've QA-deferred and will test post merge

**Review by sbishel — APPROVED** [review]

**Review by calebroseland — APPROVED** [review]

**Comment by calebroseland** [comment]
Removing DNM as UI already complies with default_team_id requirement for connection creation flow. #28841 adds default_team_id to connection accept invite flow (API and UI).

**Pull request merged**
